ElementX/Sources/Screens/RoomScreen/View/Supplementary/TimelineReadReceiptsView.swift

```diff
@@ -40,12 +40,52 @@ struct TimelineReadReceiptsView: View {
                 }
             }
             if timelineItem.properties.orderedReadReceipts.count > displayNumber {
-                let remaining = timelineItem.properties.orderedReadReceipts.count - displayNumber
                 Text("+\(remaining)")
                     .font(.compound.bodySM)
                     .foregroundColor(.compound.textPrimary)
             }
         }
+        .accessibilityElement(children: .ignore)
+        .accessibilityLabel(accessibilityLabel)
+    }
+    
+    private var remaining: Int {
+        timelineItem.properties.orderedReadReceipts.count - displayNumber
+    }
+    
+    private var accessibilityLabel: String {
+        if timelineItem.properties.orderedReadReceipts.count == 1 {
+            return L10n.a11yReadReceiptsSingle(getDisplayName(at: 0))
+        } else if timelineItem.properties.orderedReadReceipts.count <= displayNumber {
+            let limit = timelineItem.properties.orderedReadReceipts.count - 1
+            var list = ""
+            for index in 0..<limit {
+                list += "\(getDisplayName(at: index))"
+                if index != limit - 1 {
+                    list += ", "
+                }
+            }
```

Comment:
Might be a bit more succinct to do this here and below.
```suggestion
            (0..<limit).map { getDisplayName(at: $0) }.formatted(.list(type: .and, width: .narrow))
```